File: Java/src/main/java/Solution801_900/Solution804.java
package Solution801_900;

import java.util.HashSet;

/**
 * @author windows
 */
public class Solution804 {
    public int uniqueMorseRepresentations(String[] words) {
        String[] strings = {".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--", "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."};
        HashSet<String> set = new HashSet<>();
        for (int i = 0; i < words.length; i++) {
            StringBuilder sb = new StringBuilder();
            for (int j = 0; j < words[i].length(); j++) {
                sb.append(strings[words[i].charAt(j) - 'a']);
            }
            set.add(sb.toString());
        }
        return set.size();
    }


    public static void main(String[] args) {
        Solution804 solution804 = new Solution804();
    }
}

File: C++/src/Solution801_900/Solution804.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution804.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;
#include <unordered_set>

int Solution::uniqueMorseRepresentations(vector<string> &words) {
    vector<string> res = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
    unordered_set<string> set;
    for (string str : words){
        string temp;
        for (char c : str) {
            temp += res[c - 'a'];
        }
        set.insert(temp);
    }
    return set.size();
}
